rfcs/0000-nixpkgs-ifd.md

```diff
@@ -25,158 +25,171 @@ To this new generation of developers, the distro (or homebrew) is a crufty relic
-   Per the 3rd rule for Nixpkgs above, as long as we make the vendoring automatic and pure, this is fine, and improvement upon today.
-   Also, even if we didn't have the "one round of dynamism" restriction, we would still have the bootstrapping issue.
+####  Drawbacks
 
-# Alternatives
-[alternatives]: #alternatives
+- This would be a truly breaking change to nixpkgs user interface:
+  Either an additional input would need to be provided or fetched (which wouldn't interact well with restrict-eval).
 
-1. Instead of evaluating Nixpkgs twice, just evaluate `allImportedDerivations` the first round.
+- Generated code becomes a second class as the extra input would need to be optional for this reason.
+  This is problematic for central packages that use code generation already today (pandoc, cachix, …).
 
-   We could do this, and would reduce total eval time, yes.
-   But, I think it would come at the cost of inciting great controversy.
-   This means users of the `enableImportFromDerivation = false` subset of Nixpkgs would still have to *wait*, for all the IFD to complete first.
-   And remember, with mass rebuilds, that could be quite some time.
-   Increasing the critical path length of *everything* we do with Nixpkgs would cause real pain in some quarters, and I don't want that to pay that as the cost of IFD.
+- Similar Bootstrapping problems as the other alternative below: new generated code needs nixpkgs and a previous version of the generated code.
 
-   With the plan as written, users of packages depending on IFD do have to wait slightly longer as the first eval is longer (and we wait for it before beginning to build `allImportedDerivations`).
-   But I think that is fair; we would be the "new constituency", the bottom of the pecking order, and so we should be patient so that other's workflows are not disturbed.
+- `builtins.fetch*` is a nuisance to deal with at the moment and would probably need to be improved to make this work.
+  E.g. gcrooting this evaluation only dependency could prove tricky without changes to Nix.
 
-   Longer term we could revisit this, or we could e.g. double down on automatic vendoring, committing all generated code to a second "roll-up" repo.
-   Many options between those two extremes; I rather not worry to much about it now and just take the conservative polite route proposed here to begin.
+- Extra bureaucracy would be involved with updating the generated repository and the reference to it in nixpkgs.
+  Additionally, special support in CI would be required for this.
 
-2. As always, do nothing, and keep the status quo.
+### Nixpkgs itself becomes a derivation output
 
-# Unresolved questions
-[unresolved]: #unresolved-questions
+This alternative implementation was proposed by @L-as at the meeting.
+The idea is that nixpkgs would become a derivation that builds a “regular” nixpkgs source tree by augmenting files available statically with code generation.
```

Comment:
One gigantic derivation?

Wouldn't that mean a quarter-gigabyte of disk writes and ~50k inode operations for `nix-build -A hello` if I change even a single character anywhere in `nixpkgs/`?

Reply:
https://github.com/NixOS/rfcs/pull/109#discussion_r951232249